common/core/web/keyboard-processor/src/keyboards/activeLayout.ts

```diff
@@ -296,9 +303,7 @@ namespace com.keyman.keyboards {
         keys[j]['padpc']=padPercent;
 
         // compute center's default x-coord (used in headless modes)
-        (<ActiveKey> keys[j]).proportionalX = (totalPercent + padPercent + (keyPercent/2));
-        (<ActiveKey> keys[j]).proportionalWidth = keyPercent;
-
+        setProportions(keys[j] as ActiveKey, padPercent, keyPercent, totalPercent);
```

Comment:
Is this right, with `totalPercent` changing for each iteration of the loop?

Reply:
Yep.  Legacy name, though also not without a decent reason.  Note how this for-loop iterates through all but the _last_ key on the row.

That last key gets special handling, and said handling depends on the cumulative (total) percent up 'til the last key.  The branch determining how that last key gets handled is line 310 (old) / 315 (new), with the actual use of `totalPercent` found (for most cases) at line 325 (old) / 328 (new).

Reply:
The pattern of loops and such used for these calculations hasn't seen much actual change in a long, _long_ time.

For comparison, look at the equivalent code from `stable-10.0`:

https://github.com/keymanapp/keyman/blob/f16872a99896a2fc208c68ac92a827820ab07e3f/web/source/kmwosk.ts#L2331-L2358

In `activeLayout.ts` (this code file), `objectWidth` got replaced with `1`; this facilitates fat-finger calculations.  The original role of `objectWidth` is still maintained within the `VisualKeyboard` class and is used when actually building and displaying the OSK.